src/drivers/virtio/net.zig

```diff
@@ -163,7 +165,9 @@ const VirtioNet = struct {
                 }})[0..1]);
             }
 
-            self.virtio.transport.notifyQueue(self.receiveq());
+            if (rq.not_notified_num_descs > rq.num_descs / 2) {
+                self.virtio.transport.notifyQueue(self.receiveq());
+            }
```

Comment:
Only when the receive queue has more than half of the descriptors, it notifies the device it and returns the descriptors.